File: Frontend/src/app/components/interfaces/item-dto.ts
export interface ItemDto {
  type: ClothingType;
  size: ClothingSize;
  state: ItemState;
  genderSuitability: GenderSuitability;
  imageUrl: string;
  description: string;
  status: ItemStatus;
  volunteerId: number ; // Optional field, depending on the backend
}
export interface ItemDtoForEditItem {
  type: ClothingType;
  size: ClothingSize;
  state: ItemState;
  genderSuitability: GenderSuitability;
  description: string;
  status: ItemStatus;
  volunteerId: number ; // Optional field, depending on the backend
  itemId: number;
}

export enum ClothingType {
  FASHION = 'فستان',
  SHIRT = 'قميص',
  PANTS = 'بنطلون',
  JACKET = 'جاكيت'
}

export enum ClothingSize {
  SMALL = 'صغير',
  MEDIUM = 'متوسط',
  LARGE = 'كبير'
}

export enum ItemState {
  NEW = 'جديد',
  USED = 'مستعمل'
}

export enum GenderSuitability {
  MALE = 'ذكر',
  FEMALE = 'أنثى',
  UNISEX = 'مشتترك'
}

export enum ItemStatus {
  REJECTED = 'مرفوض',
  PENDING = 'معلق',
  APPROVED = 'تم_الموافقه'
}

File: Backend/src/main/java/com/ntg/backend/dto/requestDto/ItemDto.java
package com.ntg.backend.dto.requestDto;


import com.ntg.backend.entity.Item;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class ItemDto {
    private Item.ClothingType type;
    private Item.ClothingSize size;
    private Item.ItemState state;
    private Item.GenderSuitability genderSuitability;
    private String imageUrl;
    private String description;
    private Item.ItemStatus status;
    private Long volunteerId; // Optional: include volunteer ID if needed
    private Long itemId;

}
